### PR TITLE
Small Group View, Small Group Attendance, Attendance Glitch Fix

### LINKED ISSUE #184: Small Group Room Attendance
Currently, spreadsheet style attendance is a massive hassle and takes up a large majority of our small group time. If mentors could arrange attendance and attendance codes themselves on Tuesdays, this could make the process much less painful

### PR DESCRIPTION
**Pull request by seveibar**
This allows mentors to create small groups and add mentees to them. The mentees can then enter attendance codes they provide.

The frontend for mentors and students to view attendance of small groups and see their fellow mentees and projects has been added.

Currently, there isn't much administrative control over small groups- but this is the groundwork.

A bunch of utilities for accessing public user content, searching users, getting user roles was also added.

The seed was changed to remove Aaron and myself :)

The UI currently looks unremarkable, but fixing the UI shouldn't be too tricky :).

selection_054

Fixes #184 and #186

### CODE REVIEW

**Comment by seveibar** [comment]
fixes #186

File: client/app/projects/profile/profile.html [hunk 1]
```diff
@@ -9,7 +9,7 @@ <h1 ng-hide="edittingName">{{project.name}}</h1>
             </p>
             <div class="btn-group description-buttons" ng-show="userOnProject">
                 <button type="button" class='btn btn-default btn-sm' ng-click='editName()'>Edit Project Name</button>
-                <button type="button" class='btn btn-default btn-sm' ng-click='saveName()'>Save Project Name</button> 
+                <button type="button" class='btn btn-default btn-sm' ng-click='saveName()'>Save Project Name</button>
```
Comment (zlalvani) [inline client/app/projects/profile/profile.html]: is that an extra whitespace line?

File: server/api/smallgroup/index.js [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+var config = require('../../config/environment');
+var auth = require('../../auth/auth.service');
+
+var router = express.Router();
+
+// TODO permissions
+router.get('/', controller.index);
+// router.post('/',  auth.hasRole('admin'), controller.create);
+
+// router.get('/displayURP', controller.getDisplayURP);
```
Comment (kmcnellis) [inline server/api/smallgroup/index.js]: Can you clean up all this commented out code?

**Comment by kmcnellis** [comment]
When I am logged in as a non-mentor/non-admin, I can see the small group tab.  Upon clicking it, I get a failed http call to GET /api/smallgroup/undefined/members

File: client/app/smallgroup/smallgroup.controller.js [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+    }
+
+    $scope.createSmallGroup = function(){
+        $http.post("/api/smallgroup/").success(function(){
+            window.location.reload();
+            notify("Success!");
+        });
+    };
+    $scope.addUser = function(user){
+        $http.get("/api/users/search", {
```
Comment (kmcnellis) [inline client/app/smallgroup/smallgroup.controller.js]: The search is currently case sensitive  - this produces some unfriendly behavior.

File: client/app/projects/profile/profile.html (hunk 1 above)
Comment (seveibar) [inline client/app/projects/profile/profile.html]: good catch zaran

**Comment by kmcnellis** [comment]
More things that can be improved, but good enough.  I filled bugs for some improvements : #252, #254, #255, #257, #258,

**Pull request merged**